app/src/org/commcare/activities/FormEntryActivity.java

```diff
@@ -2062,7 +2062,9 @@ public boolean onScroll(MotionEvent e1, MotionEvent e2, float distanceX, float d
 
     @Override
     public void advance() {
-        next();
+        if (!questionsView.isQuestionList() && canNavigateForward()) {
```

Comment:
Heads up that Ray expects auto-advance on the last question of the form to complete the form and submit it: http://manage.dimagi.com/default.asp?186847#BugEvent.1043572
Not sure if it is worth letting him know that this is changing or ask him how important it is to his workflow